bndtools.m2e/src/bndtools/m2e/IndexConfigurator.java

```diff
@@ -108,6 +108,91 @@ public WorkspaceRepository getRepository() {
         }
     }
 
+    private final class RebuildIndexCheck extends WorkspaceJob {
+
+        private final List<IResourceChangeEvent> events = new ArrayList<>();
+        private final IMavenProjectFacade facade;
+
+        private boolean noMoreEvents;
+
+        public RebuildIndexCheck(String name, IResourceChangeEvent event, IMavenProjectFacade facade) {
+            super(name);
+            this.events.add(event);
+            this.facade = facade;
+        }
+
+        public void addEvent(IResourceChangeEvent event) {
+            synchronized (pendingJobs) {
```

Comment:
You can avoid synchronizing here since `addEvent` is only called when already synchronized on `pendingJobs`. This methods should also not be public: default access is safer.

Reply:
I agree with using default access, and I have realised that the inner class can be static. I'd like to keep the synchronized block, as it will continue to protect this method correctly if/when the IndexConfigurator is next changed/refactored.